src/components/semver/semver-directive.js

```diff
@@ -12,19 +12,35 @@
             if (!inputValue) {
               return inputValue;
             }
-
-            var r = /^(\d+)\.(\d+)\.(\d+)\.(\d+)$/;
-            if (!r.test(inputValue)) {
-              return inputValue;
-            }
-
-            // convert 4 part version to semver (1.2.3.4 to 1.2.3-4)
-            var transformedInput = inputValue.replace(r, '$1.$2.$3-$4');
-            if (transformedInput !== inputValue) {
-              modelCtrl.$setViewValue(transformedInput);
-              modelCtrl.$render();
+            
+            var z = /^(\d+)\.(\d+)\.?(\d+)?\.?(\d)?$/;
+            var result = '';
+            if (z.test(inputValue))
+            {
+              var z2 = /^(\d+)\.(\d+)$/;
+              if (z2.test(inputValue))
+                  result = inputValue.replace(z2, '$1.$2.0-0');
```

Comment:
Thanks for the pr, I'll merge this into a branch and do some testing. Can you remove the last -0 on the replaced value if it's not a 4 part version (I can do this as well)

Reply:
How do you mean? The way I structured this is that it will break as soon as possible to not process the rest (the x.x.x and x.x.x.x).